src/Autofac/ResolveRequest.cs

```diff
@@ -56,4 +56,33 @@ public ResolveRequest(Service service, ServiceRegistration serviceRegistration,
     /// Gets the component registration for the decorator target if configured.
     /// </summary>
     public IComponentRegistration? DecoratorTarget { get; }
+
+    /// <inheritdoc/>
+    public override bool Equals(object? obj) =>
+        obj is ResolveRequest other && Equals(other);
+
+    /// <inheritdoc/>
+    public bool Equals(ResolveRequest other) =>
+        Service == other.Service && Registration == other.Registration && ResolvePipeline == other.ResolvePipeline && Parameters == other.Parameters && DecoratorTarget == other.DecoratorTarget;
+
+    /// <summary>
+    /// Implements the operator ==.
+    /// </summary>
+    /// <param name="left">The left operand.</param>
+    /// <param name="right">The right operand.</param>
+    /// <returns>The result of the operator.</returns>
+    public static bool operator ==(ResolveRequest left, ResolveRequest right) => Equals(left, right);
+
+    /// <summary>
+    /// Implements the operator !=.
+    /// </summary>
+    /// <param name="left">The left operand.</param>
+    /// <param name="right">The right operand.</param>
+    /// <returns>The result of the operator.</returns>
+    public static bool operator !=(ResolveRequest left, ResolveRequest right) =>
+        !(left == right);
+
+    /// <inheritdoc/>
+    public override int GetHashCode() =>
+        Service.GetHashCode() ^ Registration.GetHashCode() ^ ResolvePipeline.GetHashCode() ^ Parameters.GetHashCode() ^ (DecoratorTarget?.GetHashCode() ?? 0);
```

Comment:
Could HashCode.Combine be used here?

Reply:
No, because this project  is compilable also for `.NET Standard`